inference-engine/src/inference_engine/src/tensor.cpp

```diff
@@ -0,0 +1,125 @@
+                                      {shape.begin(), shape.end()},
+                                      ie::TensorDesc::getLayoutByRank(shape.size())},
+                                     blob_allocator);
+    _impl->allocate();
+}
+
+Tensor::Tensor(const element::Type element_type,
+               const Shape& shape,
+               void* ptr,
+               const size_t size,
+               const Strides& strides) {
+    ie::SizeVector blk_order(shape.size());
+    std::iota(blk_order.begin(), blk_order.end(), 0);
+    ie::SizeVector dim_offset(shape.size(), 0);
+    ie::SizeVector blk_strides;
+    if (strides.empty()) {
+        blk_strides.assign(shape.begin(), shape.end());
+    } else {
+        blk_strides.assign(strides.begin(), strides.end());
+    }
+    _impl = make_blob_with_precision(ie::details::convertPrecision(element_type),
+                                     ie::TensorDesc{ie::details::convertPrecision(element_type),
+                                                    shape,
+                                                    ie::BlockingDesc{shape, blk_order, 0, dim_offset, blk_strides}},
+                                     ptr,
+                                     size);
+}
+
+Tensor::Tensor(const Tensor& owner, const Coordinate& begin, const Coordinate& end) : _so{owner._so} {
+    try {
+        _impl = owner._impl->createROI(begin, end);
+    } catch (const std::exception& ex) {
+        throw ov::Exception(ex.what());
+    } catch (...) {
+        OPENVINO_ASSERT(false, "Unexpected exception");
+    }
+}
+
+element::Type Tensor::get_element_type() const {
+    OV_TENSOR_STATEMENT(return ie::details::convertPrecision(_impl->getTensorDesc().getPrecision()));
+}
+
+void Tensor::set_shape(const ov::Shape& shape) {
+    OV_TENSOR_STATEMENT(_impl->setShape({shape.begin(), shape.end()}));
+}
+
+Shape Tensor::get_shape() const {
+    OV_TENSOR_STATEMENT({
+        auto dims = _impl->getTensorDesc().getDims();
+        return {dims.begin(), dims.end()};
+    });
+}
+
+Strides Tensor::get_strides() const {
+    OV_TENSOR_STATEMENT(return _impl->getTensorDesc().getBlockingDesc().getStrides(););
+}
+
+size_t Tensor::get_size() const {
+    OV_TENSOR_STATEMENT(return ov::shape_size(get_shape()));
+}
+
+size_t Tensor::get_byte_size() const {
+    OV_TENSOR_STATEMENT(return ov::shape_size(get_shape()) * get_element_type().size());
+}
+
+void* Tensor::data(const element::Type element_type) const {
+    OV_TENSOR_STATEMENT({
+        if (element_type != element::undefined) {
+            OPENVINO_ASSERT(element_type == get_element_type(),
+                            "Tensor element element_type should be: ",
+                            get_element_type(),
+                            " casted to ",
+                            element_type);
+        }
+        return _impl->getTensorDesc().getBlockingDesc().getOffsetPadding() * get_element_type().size() +
```

Comment:
will it work for `int4` type? can we add tests?

Reply:
added